demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +102,185 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
 
     return blobs;
 }
+
+std::string EncoderDecoderCNN::check_net_names(std::vector<std::string> output_names_encoder,
+                                               std::vector<std::string> input_names_decoder,
+                                               std::vector<std::string> output_names_decoder) {
+    if (std::find(
+            output_names_encoder.begin(),
+            output_names_encoder.end(),
+            out_enc_hidden_name_) == output_names_encoder.end())
+        return out_enc_hidden_name_;
```

Comment:
You use return values only to throw an exception later. Throw right here.

Reply:
Done

Reply:
No need to copy missing name to `name_not_exist`. Throw once you can't find a given name.